homeassistant/components/lutron_caseta/fan.py

```diff
@@ -0,0 +1,84 @@
+"""Support for Lutron Caseta fans."""
+import logging
+
+from homeassistant.components.fan import (
+    SUPPORT_SET_SPEED, FanEntity, DOMAIN,
+    SPEED_HIGH, SPEED_LOW, SPEED_MEDIUM, SPEED_OFF)
+
+from . import LUTRON_CASETA_SMARTBRIDGE, LutronCasetaDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+VALUE_TO_SPEED = {
+    'Off': SPEED_OFF,
+    'Low': SPEED_LOW,
+    'Medium': SPEED_MEDIUM,
+    'High': SPEED_HIGH
+}
+
+SPEED_TO_VALUE = {
+    SPEED_OFF: 'Off',
+    SPEED_LOW: 'Low',
+    SPEED_MEDIUM: 'Medium',
+    SPEED_HIGH: 'High'
+}
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up Lutron fan."""
+    devs = []
+    bridge = hass.data[LUTRON_CASETA_SMARTBRIDGE]
+    fan_devices = bridge.get_devices_by_domain(DOMAIN)
+
+    for fan_device in fan_devices:
+        dev = LutronCasetaFan(fan_device, bridge)
+        devs.append(dev)
+
+    async_add_entities(devs, True)
+    return True
+
+
+class LutronCasetaFan(LutronCasetaDevice, FanEntity):
+    """Representation of a Lutron Caseta fan. Including Fan Speed."""
+
+    @property
+    def speed(self) -> str:
+        """Return the current speed."""
+        return VALUE_TO_SPEED[self._state["fan_speed"]]
+
+    @property
+    def speed_list(self) -> list:
+        """Get the list of available speeds."""
+        return [SPEED_OFF, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH]
```

Comment:
We can extract this list of speeds to a module level constant which we return here.

Reply:
@MartinHjelmare list of speeds extracted to module level constant.